### PR TITLE
Title uniqueness constraint

### LINKED ISSUE #863: ATBD title naming convention
### Description
Currently there is no ATBD naming conventions but is needed. Scientists are naming completely different ATBDs the exact same or mostly the same. For example,[ Inherent Optical Properties ](https://www.earthdata.nasa.gov/apt/documents/inherent-optical-properties/v1.0) and [Inherent Optical Properties (IOPs)](https://www.earthdata.nasa.gov/apt/documents/iops/v1.0) are different ATBDs. These examples are the same scientist/science team, but this could be an issue with other science teams also producing a similar ATBD. 

One solution is to force all ATBD titles to be unique similar to how the ATBD alias currently are. However, that may bring up other issues further down the line. 

### Acceptance Criteria

- Scope out a solution
- Implement solution

### PR DESCRIPTION
**Pull request by kamicut**
Adds uniqueness constraint for ATBD title - closes #863

### CODE REVIEW

File: db/deploy/title_uniqueness.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+-- Deploy nasa-apt:title_uniqueness to pg
+-- requires: remove_reviewer_info_in_atbd_version
+
+BEGIN;
+
+ALTER TABLE apt.atbds ADD CONSTRAINT unique_title UNIQUE (title);
```
Comment (sunu) [inline db/deploy/title_uniqueness.sql:6]: Applying this migration will fail if we have existing titles that are not unique.

**Review by sunu — APPROVED** [review]
Thanks @kamicut! Changes look good to me. The only thing I'm slightly worried about is that the migration won't be applied successfully if we have existing ATBDs with duplicate titles.

File: db/deploy/title_uniqueness.sql (hunk 1 above)
Comment (kamicut) [reply db/deploy/title_uniqueness.sql:6]: Should we check this in dev and prod databases first?

**Comment by wrynearson** [comment]
@sunu @kamicut, in @thenav56's [comment](https://github.com/NASA-IMPACT/nasa-apt/issues/863#issuecomment-2157842817), he said that the uniqueness check would only be applied once the document is updated. Is that the case? I've asked @bwbaker1 to let us know once he's notified authors of documents that need to be renamed to do so.

**Comment by sunu** [comment]
> @sunu @kamicut, in @thenav56's [comment](https://github.com/NASA-IMPACT/nasa-apt/issues/863#issuecomment-2157842817), he said that the uniqueness check would only be applied once the document is updated. Is that the case? I've asked @bwbaker1 to let us know once he's notified authors of documents that need to be renamed to do so.

@wrynearson from what I understand, we won't be able to apply the uniqueness constraint introduced in the migrations if we have existing documents with duplicate titles. We have to tweak the existing duplicate titles to be unique before we can safely merge this PR.

**Comment by wrynearson** [comment]
@sunu @frozenhelium @kamicut could someone double-check that all of the ATBD titles on staging are unique before we merge this?

**Comment by frozenhelium** [comment]
@wrynearson I'll take a look

**Comment by wrynearson** [comment]
@frozenhelium checked the database. He was able to get 96 documents, all with unique ids. This should be safe to merge.

**Pull request merged**